ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestOrcFileStripeMergeRecordReader.java

```diff
@@ -63,8 +63,20 @@ public void testSplitStartsWithOffset() throws IOException {
     FileSplit split = new FileSplit(tmpPath, offset, length, (String[])null);
     OrcFileStripeMergeRecordReader reader = new OrcFileStripeMergeRecordReader(conf, split);
     reader.next(key, value);
+    // since offset is non-zero this file will not be processed.
+    Assert.assertNull(key.getInputPath());
+    split = new FileSplit(tmpPath, 0, length, (String[]) null);
```

Comment:
Offset is actually Zero here -- what am I missing?

Reply:
The code is fine but confusing, changed to 2 test classes to make it simpler.